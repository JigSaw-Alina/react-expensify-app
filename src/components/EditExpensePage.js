import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm'
import { startEditExpense, startRemoveExpense } from '../actions/expenses'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTrash } from '@fortawesome/free-solid-svg-icons'




export class EditExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.startEditExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  };
  onRemove = () => {
    this.props.startRemoveExpense({ id: this.props.expense.id });
    this.props.history.push('/');
  };
render() {
  return (
      <div>
       <div className="content-container">
        <h1>Edit Expense</h1>
      <ExpenseForm 
        expense={this.props.expense}
        onSubmit={this.onSubmit}
      />
          <button className="delete_button" onClick={this.onRemove}><FontAwesomeIcon icon={faTrash} color="white" size="2x"/></button>
      </div>
    </div>
   );
  };
};



const mapStateToProps = (state, props) =>  ({
    expense: state.expenses.find((expense) => expense.id === props.match.params.id )
  });

const mapDispatchToProps = (dispatch, props) => ({
  startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
    startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
});


export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
