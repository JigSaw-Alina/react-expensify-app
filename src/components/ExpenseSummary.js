import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expensesTotal';
import numeral from 'numeral';



export const ExpenseSummary = ({expenseCount, expensesTotal}) => {
    const formatExpenseWord = expenseCount === 1 ? 'expense' : 'expenses';
    const formatexpensesTotal = numeral(expensesTotal).format('$0,0');
    return (
       <div className="page-header">
        <div className="expense_button">
          <Link id="new_expense" to="/create" >Create Expense</Link>
        </div>
        <div className="content-container">
         <h1 className="page-header__title">Viewing <span className="expenses_summary">{expenseCount}</span> {formatExpenseWord} totalling <span className="expenses_summary">{formatexpensesTotal}</span></h1>  
        </div>
      </div>
    );
}


const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters)
    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
   }
}


export default  connect(mapStateToProps)(ExpenseSummary);

