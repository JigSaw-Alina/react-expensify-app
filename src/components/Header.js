import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSignOutAlt, faUserCircle } from '@fortawesome/free-solid-svg-icons'


export const Header = ({ startLogout }) => (
  <header>
  <div className="container__content_page">
    <div className="box_content__left__container">
      <div id="user__icon">
        <FontAwesomeIcon icon={faUserCircle} color="orange" size="5x"/>
      </div>
      <div>
        <button onClick={startLogout}><FontAwesomeIcon icon={faSignOutAlt} size="2x"/></button>
      </div>
        
      <Link to="/dashboard"><img className="box-content__title" src="/images/expensify-title.png" /></Link>
    </div>
  </div>
  </header>
);

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
});

export default connect(undefined,mapDispatchToProps)(Header);





// <NavLink to="/help" activeClassName="is-active">Help</NavLink> 