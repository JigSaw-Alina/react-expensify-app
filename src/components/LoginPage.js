import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faGlobe, faComment, faAddressBook, faConciergeBell } from '@fortawesome/free-solid-svg-icons'



export const LoginPage = ({ startLogin }) => (
    <div className="container__layout_page">
    <div className="box_layout__left__container">
     <div className="box_layout">
      <img className="box-layout__title" src="/images/expensify-title.png" />
      <span id="concierge__expensify">
      <FontAwesomeIcon icon={faConciergeBell} color="white" size="2x"/>
      </span>
      <p id="welcome__expensify">Welcome! How would you like to connect?</p>
      <div className ="button_container">
      <button className="buttons__nav" onClick={startLogin}>
      <span id="google__icon">
      <FontAwesomeIcon icon={faGlobe}/>
      </span></button>
      <button className="buttons__nav" >
      <span id="facebook__icon">
      <FontAwesomeIcon icon={faAddressBook}/>
      </span></button>
      <button className="buttons__nav">
      <span id="phone_number__icon">
      <FontAwesomeIcon icon={faComment}/>
      </span></button>
      </div>
      <p id="link__expensify">By logging in, you agree to the <a href="" target="_blank">privacy policy.</a></p>
    </div>
    </div>

    <div className="box_layout__right__container">
        <h1 className="quotes__expensify">Don't just get paid back.<br />Pay it forward</h1>
        <div id="vl__expensify"/>
        <p id="note__one">Every time you book a hotel,<br /> you cover a kid's school lunch</p>
    </div>
    </div>
);


const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);

