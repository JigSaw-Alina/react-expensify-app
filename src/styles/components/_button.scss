.button_container {
    display: -ms-grid;
    display: grid;
   grid-row-gap: 1.5rem;
};

#google__icon {
    display: inline-block;
    text-indent: -35rem;
    font-family: $login-page-font;
    font-weight: bold;
    &::after {
      content: " Google";
      display: inline;
      padding: 0 0 0 0.6rem;
    }
};


#facebook__icon {
    display: inline-block;
    text-indent: -33rem;
    font-family: $login-page-font;
    font-weight: bold;
    &::after {
      content: " Facebook";
      display: inline;
      padding: 0 0 0 0.6rem;
    }
};


#phone_number__icon {
    display: inline-block;
    text-indent: -29rem;
    font-family: $login-page-font;
    font-weight: bold;
    &::after {
      content: " Phone Number";
      display: inline;
      padding: 0 0 0 0.6rem;
    }
};


.buttons__nav {
    border-radius: $l-size;
    padding: 1.9rem 4rem 1.9rem 17.9rem;
    background: $-white;
    color: #43525d;
    margin: 0 5.3rem;
    border: none;
    &:hover {
        border-style: solid;
        border-color: #19B5FE;
    }
};


#concierge__expensify {
    padding: 30rem 0 0 8.2rem;
    color:$-white;
    &::after {
      content: ' Concierge';
      padding: 0 0 0 0.6rem;
      font-weight: bold;
    }
};


#welcome__expensify {
    padding: 0 0 1rem 7rem;
    color:$-white;
    font-family: $login-page-font;
    font-weight: bold;
};

#link__expensify {
    padding: 20rem 0 0 9rem;
    color:$-white;
    font-family: $login-page-font;
    font-size: 1.2rem;
};

