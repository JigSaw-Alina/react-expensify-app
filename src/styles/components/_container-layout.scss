.container__layout_page {
    display: grid;
    grid-template-columns: repeat(7, 1fr);
    grid-template-rows: 75rem;
};

.box_layout__left__container {
    background-color:#43525d;
    grid-column: 1 / 1;
};

.box_layout__right__container {
    grid-column: 2 / 8;
};


.box-layout__title {
    width: 27rem;
    height: $xlll-size;
    padding: $xll-size 0 0 $xlll-size;
    margin: 0 0 85px 0;
};

.box-layout__google {
    width: 20px;
    height: 20px;
    padding: 0 10px 0 0;
    margin: 0;
  
}

.box-layout__fb {
    width: 23px;
    height: 23px;
    padding: 0 11px 0 10px;
    margin: 0;
}

.box-layout__msg {
    width: 26px;
    height: 26px;
    margin: 0;
    padding: 0 20px 0 0;
}

.box-layout__concierge {
    width: 25rem;
    height: $xlll-size;
}


.box_layout__left__container {
    background-color: #43525d;
    grid-column: 1 / 3 ;
   
};

.box_layout__right__container {
    color:$-white;
    background: url('/images/bg.jpg') no-repeat center;
    background-size: cover;
    grid-column: 3 / 8 ;
    padding: 50.5rem 0 0 8.4rem;
}

#vl__expensify {
    border-bottom: 0.5rem solid #ffff00;
    width: 3.6rem;
}

