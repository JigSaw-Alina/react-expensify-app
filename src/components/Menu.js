import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faBars } from '@fortawesome/free-solid-svg-icons';
import Header from './Header';

class Menu extends React.Component {
  constructor(props){
      super(props)
    this.state = {
        isSidebarOpen: false
    };
  };

  handleMenu = () => {
    this.setState(() => ({ isSidebarOpen: !this.state.isSidebarOpen }));
};

  render() {
      const { isSidebarOpen } = this.state
    return (
     <div className="container">
      <div className="menu-button" onClick={this.handleMenu}>
        <FontAwesomeIcon  icon={faBars} color="black" size="2x"/>
      </div>   
	{/*	Sidebar */}
        {<Header /> && (
            <div>
            
            </div>
        )}
      
    </div>      
     )
  }; 
};

export default Menu;


