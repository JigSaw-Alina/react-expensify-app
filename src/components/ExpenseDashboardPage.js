import React from 'react';
import ExpenseList from './ExpenseList'
import ExpsenListFilters from './ExpenseListFilters';
import ExpenseSummary from './ExpenseSummary';
import Menu from './Menu';




const ExpenseDashboardPage = () => (
  <div>
    <ExpenseSummary />
    <ExpsenListFilters />
    <ExpenseList />
    </div>
);

export default ExpenseDashboardPage;
