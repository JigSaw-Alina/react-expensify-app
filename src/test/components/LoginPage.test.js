import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage }  from '../../components/LoginPage';

test('should correctly render LoginPage', () => {
  const wrapper = shallow(<LoginPage />);
  expect(wrapper).toMatchSnapshot();
});


//LoginPage test file -> Should call startLogin on button click
test('should call LoginPage on button click', () => {
  const startLogin = jest.fn();
  const wrapper = shallow(<LoginPage startLogin={startLogin} />);
  wrapper.find('button').simulate('click');
  expect(startLogin).toHaveBeenCalled();
});

