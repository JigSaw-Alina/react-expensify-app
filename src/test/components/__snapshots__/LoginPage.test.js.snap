// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render LoginPage 1`] = `
<div
  className="container__layout_page"
>
  <div
    className="box_layout__left__container"
  >
    <div
      className="box_layout"
    >
      <img
        className="box-layout__title"
        src="/images/expensify-title.png"
      />
      <span
        id="concierge__expensify"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          color="white"
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f562",
                "M288 130.54V112h16c8.84 0 16-7.16 16-16V80c0-8.84-7.16-16-16-16h-96c-8.84 0-16 7.16-16 16v16c0 8.84 7.16 16 16 16h16v18.54C115.49 146.11 32 239.18 32 352h448c0-112.82-83.49-205.89-192-221.46zM496 384H16c-8.84 0-16 7.16-16 16v32c0 8.84 7.16 16 16 16h480c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16z",
              ],
              "iconName": "concierge-bell",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="2x"
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </span>
      <p
        id="welcome__expensify"
      >
        Welcome! How would you like to connect?
      </p>
      <div
        className="button_container"
      >
        <button
          className="buttons__nav"
        >
          <span
            id="google__icon"
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    496,
                    512,
                    Array [],
                    "f0ac",
                    "M336.5 160C322 70.7 287.8 8 248 8s-74 62.7-88.5 152h177zM152 256c0 22.2 1.2 43.5 3.3 64h185.3c2.1-20.5 3.3-41.8 3.3-64s-1.2-43.5-3.3-64H155.3c-2.1 20.5-3.3 41.8-3.3 64zm324.7-96c-28.6-67.9-86.5-120.4-158-141.6 24.4 33.8 41.2 84.7 50 141.6h108zM177.2 18.4C105.8 39.6 47.8 92.1 19.3 160h108c8.7-56.9 25.5-107.8 49.9-141.6zM487.4 192H372.7c2.1 21 3.3 42.5 3.3 64s-1.2 43-3.3 64h114.6c5.5-20.5 8.6-41.8 8.6-64s-3.1-43.5-8.5-64zM120 256c0-21.5 1.2-43 3.3-64H8.6C3.2 212.5 0 233.8 0 256s3.2 43.5 8.6 64h114.6c-2-21-3.2-42.5-3.2-64zm39.5 96c14.5 89.3 48.7 152 88.5 152s74-62.7 88.5-152h-177zm159.3 141.6c71.4-21.2 129.4-73.7 158-141.6h-108c-8.8 56.9-25.6 107.8-50 141.6zM19.3 352c28.6 67.9 86.5 120.4 158 141.6-24.4-33.8-41.2-84.7-50-141.6h-108z",
                  ],
                  "iconName": "globe",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              swapOpacity={false}
              symbol={false}
              title=""
              transform={null}
            />
          </span>
        </button>
        <button
          className="buttons__nav"
        >
          <span
            id="facebook__icon"
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    448,
                    512,
                    Array [],
                    "f2b9",
                    "M436 160c6.6 0 12-5.4 12-12v-40c0-6.6-5.4-12-12-12h-20V48c0-26.5-21.5-48-48-48H48C21.5 0 0 21.5 0 48v416c0 26.5 21.5 48 48 48h320c26.5 0 48-21.5 48-48v-48h20c6.6 0 12-5.4 12-12v-40c0-6.6-5.4-12-12-12h-20v-64h20c6.6 0 12-5.4 12-12v-40c0-6.6-5.4-12-12-12h-20v-64h20zm-228-32c35.3 0 64 28.7 64 64s-28.7 64-64 64-64-28.7-64-64 28.7-64 64-64zm112 236.8c0 10.6-10 19.2-22.4 19.2H118.4C106 384 96 375.4 96 364.8v-19.2c0-31.8 30.1-57.6 67.2-57.6h5c12.3 5.1 25.7 8 39.8 8s27.6-2.9 39.8-8h5c37.1 0 67.2 25.8 67.2 57.6v19.2z",
                  ],
                  "iconName": "address-book",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              swapOpacity={false}
              symbol={false}
              title=""
              transform={null}
            />
          </span>
        </button>
        <button
          className="buttons__nav"
        >
          <span
            id="phone_number__icon"
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    512,
                    512,
                    Array [],
                    "f075",
                    "M256 32C114.6 32 0 125.1 0 240c0 49.6 21.4 95 57 130.7C44.5 421.1 2.7 466 2.2 466.5c-2.2 2.3-2.8 5.7-1.5 8.7S4.8 480 8 480c66.3 0 116-31.8 140.6-51.4 32.7 12.3 69 19.4 107.4 19.4 141.4 0 256-93.1 256-208S397.4 32 256 32z",
                  ],
                  "iconName": "comment",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              swapOpacity={false}
              symbol={false}
              title=""
              transform={null}
            />
          </span>
        </button>
      </div>
      <p
        id="link__expensify"
      >
        By logging in, you agree to the 
        <a
          href=""
          target="_blank"
        >
          privacy policy.
        </a>
      </p>
    </div>
  </div>
  <div
    className="box_layout__right__container"
  >
    <h1
      className="quotes__expensify"
    >
      Don't just get paid back.
      <br />
      Pay it forward
    </h1>
    <div
      id="vl__expensify"
    />
    <p
      id="note__one"
    >
      Every time you book a hotel,
      <br />
       you cover a kid's school lunch
    </p>
  </div>
</div>
`;
